airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -873,6 +874,11 @@ def dry_run(self) -> None:
         pod = self.build_pod_request_obj()
         print(yaml.dump(prune_dict(pod.to_dict(), mode="strict")))
 
+    def progress_callback(self) -> None:
+        """
+        Expose a callback hook on status lines
+        """
+        pass
```

Comment:
Instead of providing an empty function that needs to be overridden by subclassing, I wonder if this should instead be an argument to the operator. So users can do something like

```python
def progress_callback(message):
    ....

t1 = KubernetesPodOperator(
    ...,
    progress_callback=progress_callback,
)
```